tools/coqdoc/cpretty.mll

```diff
@@ -504,9 +504,9 @@ rule coq_bol = parse
       { Lexing.new_line lexbuf; begin_show (); coq_bol lexbuf }
   | space* end_show nl
       { Lexing.new_line lexbuf; end_show (); coq_bol lexbuf }
-  | space* begin_details nl
-      { Lexing.new_line lexbuf;
-        let s = details_body lexbuf in
+  | space* begin_details (* the [nl] here is embedded into the call of [details_body] *)
```

Comment:
If I understand correctly, there shouldn't be a newline anyway. Rather, the gotcha is that the comment is still open, and we close it after parsing the summary in `details_body`.

Reply:
The newline is checked by `details_body`… although now that I think about it, it is `nl?` rather than `nl` so maybe we should rework that. Will make some test to that end.

Reply:
Right, after closing the comment. I was referring to the fact that if you put `nl` right here, it would expect a newline before the colon, which is incorrect and the cause for the issue: `(* begin details \n : summary *)`.

Reply:
You are totally right!

Reply:
I still think that the important bit of information to mention in this comment is that the comment is still open at that point, and that `details_body` closes it.